File: pdu/python/std_msgs/pdu_cdr_conv_Empty.py
from .pdu_pytype_Empty import Empty
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Empty(writer: CdrWriter, src: Empty):
    writer.write_uint8(src.hako_dummy)


def cdr_body_to_py_Empty(reader: CdrReader, dst: Empty):
    dst.hako_dummy = reader.read_uint8()
    return dst


def py_to_cdr_Empty(src: Empty) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Empty(writer, src)
    return writer.bytes()


def cdr_to_py_Empty(cdr_payload) -> Empty:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Empty()
    return cdr_body_to_py_Empty(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Empty.js
import { Empty } from './pdu_jstype_Empty.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduEmptyConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Empty } src
     */
    static to_cdr_body(writer, src) {
        return;
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Empty } dst
     * @returns { Empty }
     */
    static cdr_body_to_js(reader, dst) {
        return dst;
        return dst;
    }

    /**
     * @param { Empty } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Empty }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Empty());
    }
}
